napari/viewer.py

```diff
@@ -103,27 +103,28 @@ def update_console(self, variables):
         else:
             self.window.qt_viewer.console.push(variables)
 
-    def screenshot(self, path=None, *, with_viewer=False):
+    def screenshot(self, path=None, *, canvas_only=False):
```

Comment:
I really like `canvas_only`. I'm ok with this now being false by default, but I know previously @jni had expressed desires for it to be `True` by default, and so I just want to make sure he's ok with this change.

For the nbscreenshot method I like `False` by default, so it might be then strange if our two screenshot functions had different defaults.

Reply:
Just btw this is a breaking API change. Maybe we want to keep `with_viewer` around for a couple of versions with a warning? Or with a ValueError pointing them to the new kwarg?

Reply:
I feel like most user won't be using `with_viewer` they will have nothing, and now they will need to add the `canvas_only` keyword to preserve the previous behavior. We might have to ask @guiwitz to update [naparimovie](https://github.com/guiwitz/naparimovie) to add `canvas_only=True` and pin to `napari>=0.3.2` until we get #780 merged otherwise `naparimovie` will break after this PR gets merged.

Reply:
and by break, i mean now include the napari viewer frame by default

Reply:
No problem I'll update naparimovie for this. Thanks @GenevieveBuckley for this functionality which will be very helpful for demos/teaching! Do I correctly understand that we are going from having a default ```screenshot``` with no napari-viewer to a default with it? Isn't that ```screenshot``` function the only way for people to grab an output from napari that is usable e.g. to make a figure? In that case I don't think most people want the canvas by default. Except those using screenshots for demos/teaching, which now can use the ```nbscreenshot``` function which could have a default with napari-viewer.

Reply:
It's @jni you need to thank, @guiwitz!

It was his idea, and this PR is virtually indistinguishable from his initial implementation. There's only one or two tiny changes here (`_repr_png_` lives in it's own object class now, not on the viewer class, and I removed the scikit-image dependency)

It's really nice, I like it a lot too. Thanks @jni

Reply:
Aw shucks. 😊 Thanks for modernising it @GenevieveBuckley!

I agree with @guiwitz that `.screenshot()` should default to `canvas_only=True` and `nbscreenshot()` to `canvas_only=False`. They are not the same function, but rather serve very different purposes, so different defaults make sense, imho. What do you think @sofroniewn @GenevieveBuckley?

Reply:
I am fine with those two functions having different defaults. Agreed they serve different purposes. @GenevieveBuckley if you agree do you want to change the `.screenshot()` default to `canvas_only=True`, and then I will merge this!! And I don't think @guiwitz will actually need to update `naparimovie` as a quick search of the code turns up no hits for `with_viewer`. Probably indicates we're making the right choice here